File: src/main/java/xyz/aeolia/lib/command/admin/util/StatusToggleHandler.java
package xyz.aeolia.lib.command.admin.util;

import org.bukkit.command.CommandSender;
import org.jetbrains.annotations.NotNull;
import xyz.aeolia.lib.manager.StatusManager;
import xyz.aeolia.lib.sender.MessageSender;

import java.util.List;

import static xyz.aeolia.lib.utils.Message.Generic.COMMAND_USAGE;
import static xyz.aeolia.lib.utils.Message.Generic.TOO_MANY_ARGS;

public class StatusToggleHandler extends SubCommandHandler {
  final String instance;
  public StatusToggleHandler(String instance) {
    this.instance = switch (instance) {
      case "roe" -> "restartonempty";
      case "lc" -> "lockchat";
      default -> instance;
    };
  }

  @Override
  public boolean handle(@NotNull CommandSender sender, String @NotNull [] args) {
    String instanceFormatted = switch (instance) {
      case "restartonempty" -> "RestartOnEmpty";
      case "lockchat" -> "Chat lock";
      default -> "Unknown instance";
    };
    if (args.length > 0) {
      if (args.length > 1) {
        MessageSender.sendMessage(sender, TOO_MANY_ARGS, true);
        MessageSender.sendMessage(sender, "/util " + instance + " [true/false/status].", true);
      }
      switch (args[0].toLowerCase()) {
        case "true":
          StatusManager.setStatus(instance, true);
          MessageSender.sendMessage(sender, instanceFormatted + " enabled.", true);
          return true;
        case "false":
          StatusManager.setStatus(instance, false);
          MessageSender.sendMessage(sender, instanceFormatted + " disabled.", true);
          return true;
        case "status":
          MessageSender.sendMessage(sender,
                  instanceFormatted + " " + (StatusManager.getStatus(instance) ? "enabled" : "disabled") + ".",
                  true);
          return true;
        default:
          MessageSender.sendMessage(sender, COMMAND_USAGE, true);
          return false;
      }
    }
    boolean toSet = !StatusManager.getStatus(instance);
    StatusManager.setStatus(instance, toSet);
    MessageSender.sendMessage(sender, instanceFormatted + " is " + (toSet ? "enabled" : "disabled") + ".", true);
    return true;
  }

  @Override
  public @NotNull List<@NotNull String> getAlias() {
    return List.of("roe", "restartonempty", "lc", "lockchat");
  }
}

File: src/main/kotlin/xyz/aeolia/lib/command/admin/util/StatusToggleHandler.kt
package xyz.aeolia.lib.command.admin.util

import org.bukkit.command.CommandSender
import xyz.aeolia.lib.data.SubCommandHandler
import xyz.aeolia.lib.manager.StatusManager
import xyz.aeolia.lib.sender.MessageSender.sendMessage
import xyz.aeolia.lib.utils.Message.Generic.COMMAND_USAGE
import xyz.aeolia.lib.utils.Message.Generic.TOO_MANY_ARGS

class StatusToggleHandler(instance: String) : SubCommandHandler() {

  override val alias = listOf("roe", "restartonempty", "lc", "lockchat")

  val instance: String = when (instance) {
    "roe" -> "restartonempty"
    "lc" -> "lockchat"
    else -> instance
  }

  override fun handle(sender: CommandSender, args: Array<out String>): Boolean {
    val instanceFormatted = when (instance) {
      "restartonempty" -> "RestartOnEmpty"
      "lockchat" -> "Chat lock"
      else -> "Unknown instance"
    }

    if (args.isEmpty()) return true.also {
      val toSet = !StatusManager.getStatus(instance)
      StatusManager.setStatus(instance, toSet)
      sendMessage(
        sender,
        instanceFormatted + " is " + (if (toSet) "enabled" else "disabled") + ".", true
      )
    }

    if (args.size > 1) return true.also {
      sendMessage(sender, TOO_MANY_ARGS, true)
      sendMessage(sender, "/util $instance [true/false/status].", true)
    }

    when (args[0].lowercase()) {
      "true" -> return true.also {
        StatusManager.setStatus(instance, true)
        sendMessage(sender, "$instanceFormatted enabled.", true)
      }

      "false" -> return true.also {
        StatusManager.setStatus(instance, false)
        sendMessage(sender, "$instanceFormatted disabled.", true)
      }

      "status" -> return true.also {
        sendMessage(
          sender,
          instanceFormatted + " " + (if (StatusManager.getStatus(instance)) "enabled" else "disabled") + ".",
          true
        )
      }

      else -> return false.also { sendMessage(sender, COMMAND_USAGE, true) }
    }
  }
}